pkg/controller/job/job_controller_test.go

```diff
@@ -6756,19 +6765,49 @@ func TestSyncOrphanPod(t *testing.T) {
 					}
 				})
 			}
+			var orphanPod *v1.Pod
+			var orphanKey orphanPodKey
+			if tc.podSelector != nil {
+				podBuilder := buildPod().name(name).labels(tc.podSelector.MatchLabels).deletionTimestamp().trackingFinalizer()
+				if tc.owner != nil {
+					podBuilder = podBuilder.owner(*tc.owner)
+				}
+				orphanPod = podBuilder.Pod
+				orphanPod, err = clientset.CoreV1().Pods("default").Create(ctx, orphanPod, metav1.CreateOptions{})
+				if err != nil {
+					t.Fatalf("Creating orphan pod with label selector: %v", err)
+				}
 
-			podBuilder := buildPod().name(name).deletionTimestamp().trackingFinalizer()
-			if tc.owner != nil {
-				podBuilder = podBuilder.owner(*tc.owner)
-			}
-			orphanPod := podBuilder.Pod
-			orphanPod, err := clientset.CoreV1().Pods("default").Create(ctx, orphanPod, metav1.CreateOptions{})
-			if err != nil {
-				t.Fatalf("Creating orphan pod: %v", err)
+				// Simulate syncing using pod selector
+				selector, err := metav1.LabelSelectorAsSelector(tc.podSelector)
+				if err != nil {
+					t.Fatalf("Error parsing pod label selector: %v", err)
+				}
+				orphanKey = orphanPodKey{
+					kind:      OrphanPodKeyKindSelector,
+					namespace: orphanPod.Namespace,
+					value:     selector.String(),
+				}
+			} else {
+				podBuilder := buildPod().name(name).deletionTimestamp().trackingFinalizer()
+				if tc.owner != nil {
+					podBuilder = podBuilder.owner(*tc.owner)
+				}
+				orphanPod = podBuilder.Pod
+				orphanPod, err = clientset.CoreV1().Pods("default").Create(ctx, orphanPod, metav1.CreateOptions{})
+				if err != nil {
+					t.Fatalf("Creating orphan pod: %v", err)
+				}
+				orphanKey = orphanPodKey{
+					kind:      OrphanPodKeyKindName,
+					namespace: orphanPod.Namespace,
+					value:     orphanPod.Name,
+				}
 			}
-			err = manager.syncOrphanPod(ctx, cache.MetaObjectToName(orphanPod).String())
+			// Sync orphan pod by selector
+			err = manager.syncOrphanPod(ctx, orphanKey)
 			if err != nil {
-				t.Fatalf("Failed sync orphan pod: %v", err)
+				t.Fatalf("Failed to sync orphan pod by label selector: %v", err)
```

Comment:
```suggestion
				t.Fatalf("Failed to sync orphan pod: %v", err)
```
nit, since it can be either name or selector.